scimma_admin/hopskotch_auth/views.py

```diff
@@ -404,21 +409,24 @@ def create_topic(request):
     # if the requesting user is a member of the proposed owning group it also proves that the 
     # group exists, so we don't need to check that separately
 
-    if not "topic_name" in request.POST or not validate_topic_name(request.POST["topic_name"]):
-        return redirect_with_error(request, "Create a topic", 
-                                   "Missing or invalid topic name", 
+    group = Group.objects.get(id=group_id)
+
+    if not "topic_name" in request.POST or len(request.POST["topic_name"]) == 0:
+        return redirect_with_error(request, "Create a topic", "Missing topic name",
+                                   reverse("edit_group")+"?group_id="+group_id)
+
+    topic_name = group.name+"."+request.POST["topic_name"]
+
+    if not validate_topic_name(topic_name):
```

Comment:
I think you want to validate it before you add in `group.name` and `.`.

Reply:
It needs to be as it is to ensure that the final thing we intend to hand to Kafka meets Kafka's rules (which `validate_topic_name` should already implement). Note that `validate_topic_name` is an existing function with some deliberate differences from `validate_group_name`.

Reply:
Ah, right, yes - I misread and saw them as the same name.